packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -933,41 +965,106 @@ class DropdownButtonFormField<T> extends FormField<T> {
   /// The [DropdownButton] [items] parameters must not be null.
   DropdownButtonFormField({
     Key key,
-    T value,
-    @required List<DropdownMenuItem<T>> items,
+    this.value,
+    @required this.items,
+    this.hint,
     this.onChanged,
-    InputDecoration decoration = const InputDecoration(),
+    this.decoration = const InputDecoration(),
     FormFieldSetter<T> onSaved,
     FormFieldValidator<T> validator,
-    Widget hint,
+    bool autovalidate = false,
```

Comment:
Sorry, I hadn't noticed this before: why are we creating all of these fields? These parameters are all just getting passed along to the super call; there doesn't seem to be any need to make them fields of DropDownButtonFormField.

Reply:
Sorry @HansMuller , couldn't get you. If you donot create those fields, how are we going to pass parameteres from DropdownButtonFormField to DropdownButton and FormField. 😕

Reply:
I think what he means here is that you can just do: 
```
    Widget disabledHint,
    int elevation = 8,
    // rest of properties
    double iconSize = 24.0,
    bool isDense = false,
    bool isExpanded = false,
```
and remove all of the properties on `DropdownButtonFormField` itself. If `DropdownButtonFormField` is not directly using any of these properties and they're just being passed down to `DropdownButton`, then there's no need to define them here.

Reply:
If that's possible, that helps us avoid duplicating the properties, as well as having to template out the docs as you already have.

Reply:
That's right: all of the new values are being passed along via the `super()` call, there's no need to squirrel them away in this class.

Reply:
@HansMuller @shihaohong Is this ok ?